automation_tools/__init__.py

```diff
@@ -420,14 +420,10 @@ def setup_default_libvirt(bridge=None, ip_address="192.168.100.1"):
     """
     run('yum install -y libvirt libvirt-daemon-kvm virt-install qemu-kvm')
     run('lsmod | grep kvm_', warn_only=True)
-    run('sed -i \'s/^#*\s*LIBVIRTD_ARGS=.*/LIBVIRTD_ARGS=--listen/\''
-        ' /etc/sysconfig/libvirtd')
-    run('sed -i \'s/^#*\s*listen_tls\s*=.*/listen_tls = 0/\''
-        ' /etc/libvirt/libvirtd.conf')
-    run('sed -i \'s/^#*\s*listen_tcp\s*=.*/listen_tcp = 1/\''
-        ' /etc/libvirt/libvirtd.conf')
-    run('sed -i \'s/^#*\s*auth_tcp\s*=.*/auth_tcp = "none"/\''
-        ' /etc/libvirt/libvirtd.conf')
+    run(r'sed -i \'s/^#*\s*LIBVIRTD_ARGS=.*/LIBVIRTD_ARGS=--listen/\' /etc/sysconfig/libvirtd')
```

Comment:
(sorry for the drive by, I'm writing a tool to [automatically fix these](https://github.com/asottile/pyupgrade/issues/47))

there's a subtle difference here between the left hand side and the right hand side:

```pycon
>>> x = 'sed -i \'s/^#*\s*LIBVIRTD_ARGS=.*/LIBVIRTD_ARGS=--listen/\''
>>> y = r'sed -i \'s/^#*\s*LIBVIRTD_ARGS=.*/LIBVIRTD_ARGS=--listen/\''
>>> print(x)
sed -i 's/^#*\s*LIBVIRTD_ARGS=.*/LIBVIRTD_ARGS=--listen/'
>>> print(y)
sed -i \'s/^#*\s*LIBVIRTD_ARGS=.*/LIBVIRTD_ARGS=--listen/\'
```

the `\'` inside a raw string is interpreted literally *and* escapes the quote.

Reply:
yw and thx

so I better escape instead of 'r'

Reply:
resolved